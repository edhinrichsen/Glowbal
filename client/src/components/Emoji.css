.emoji {
  /* Strange hack that centers emojis on retina displays 🤷‍♂️ */
  display: inline-block;
  font-size: 4em;
  transform: scale(0.25) translateY(1.1em);
  margin: -1em -0.333em;
}
