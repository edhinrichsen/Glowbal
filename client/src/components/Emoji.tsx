import React from 'react';

type EmojiProps = {
  /** Aria-label for accessibility */
  label?: string;
  /** Emoji symbol to display e.g. ðŸš€ */
  symbol: string;
};

function Emoji(props: EmojiProps) {
  return (
    <span
      className="emoji"
      role="img"
      aria-label={props.label ? props.label : undefined}
      aria-hidden={props.label ? undefined : true}
    >
      {props.symbol}
    </span>
  );
}

export default Emoji;
