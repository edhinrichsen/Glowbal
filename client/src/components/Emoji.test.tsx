import React from 'react';
import { shallow, mount } from 'enzyme';
import Emoji from './Emoji';

describe('rendering Emoji component', () => {

  it('renders without crashing', () => {
    const component = shallow(<Emoji symbol="ðŸ™‚" />);
    expect(component).toMatchSnapshot();
  });

});
  
describe('Emoji component accessibility', () => {

  it('sets aria-hidden when no label is passed', () => {
    const component = mount(<Emoji symbol="ðŸ™‚" />);
    expect(component.find({ className: "emoji" }).prop('aria-hidden')).toBe(true);
  });

  it('sets aria-hidden when no label is passed', () => {
    const component = mount(<Emoji symbol="ðŸ™‚" />);
    expect(component.find({ className: "emoji" }).prop('aria-hidden')).toBe(true);
    expect(component.find({ className: "emoji" }).prop('aria-label')).toBe(undefined);
  });

  it('sets aria-hidden when a label is passed', () => {
    const component = mount(<Emoji symbol="ðŸ™‚" label="testing label" />);
    expect(component.find({ className: "emoji" }).prop('aria-hidden')).toBe(undefined);
    expect(component.find({ className: "emoji" }).prop('aria-label')).toBe("testing label");
  });

  it('sets role as img', () => {
    const component = mount(<Emoji symbol="ðŸ™‚" />);
    expect(component.find({ className: "emoji" }).prop('role')).toBe("img");
  });

});

