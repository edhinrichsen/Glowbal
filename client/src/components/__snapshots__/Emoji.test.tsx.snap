// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering Emoji component renders without crashing 1`] = `
<span
  aria-hidden={true}
  className="emoji"
  role="img"
>
  ðŸ™‚
</span>
`;
